File: src/data/Model/Cms360/Registration/Enrollments.cs
/*
*   Auther: H.Muhammad Kamran
*   email: hmuhdkamran@gmail.com
*   contact: +92 (313 / 333) 9112 845
*/

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cms360.Data.Model
{
    [Table("Enrollments", Schema = "Registration")]
    public partial class RegistrationEnrollments
    {
        [Key]
        [Required]
        public Guid EnrollmentId { get; set; }
        [Required]
        public Guid AdmissionFormId { get; set; }
        [Required]
        public Guid[] SectionCourseLinkId { get; set; }
        [Required]
        public int StatusId { get; set; }
        [Required]
        public Guid LoggerId { get; set; }

    }

    [Table("VWStudentToEnroll", Schema = "Registration")]
    public partial class StudentToEnrollVM
    {
        [Key]
        public Guid AdmissionFormId { get; set; }
        public Guid CampusProgramId { get; set; }
        public Guid ClassId { get; set; }

        public Guid StudentId { get; set; }
        public Guid AdmissionTypeId { get; set; }
        public string RollNo { get; set; }
        public string RefferenceNo { get; set; }

        public string FullName { get; set; }
        public string FatherName { get; set; }
        public string StudentCNIC { get; set; }
        public Guid SectionId { get; set; }
        public string Range { get; set; }
        public bool IsSelected { get; set; }
        public bool IsDisabled { get; set; }
        public Guid GenderId { get; set; }
        public int StatusId { get; set; }

        public string ObtainedMarks { get; set; }
    }
    [Table("VWStudentsToEnrolledPercentage", Schema = "Registration")]
    public partial class StudentsToEnrolledPercentageVM
    {
        [Key]
        public Guid AdmissionFormId { get; set; }
        public Guid CampusProgramId { get; set; }
        public Guid ClassId { get; set; }

        public Guid StudentId { get; set; }
        public Guid AdmissionTypeId { get; set; }
        public string RollNo { get; set; }
        public string RefferenceNo { get; set; }

        public string FullName { get; set; }
        public string FatherName { get; set; }
        public string StudentCNIC { get; set; }
        public Guid SectionId { get; set; }
        public string Range { get; set; }
        public bool IsSelected { get; set; }
        public bool IsDisabled { get; set; }
        public Guid GenderId { get; set; }
        public int StatusId { get; set; }
        public decimal Percentage { get; set; }
    }
    [Table("VWStudentEnrolled", Schema = "Registration")]
    public partial class StudentEnrolledVM
    {

        public Guid AdmissionFormId { get; set; }
        public Guid CampusProgramId { get; set; }
        public Guid ClassId { get; set; }

        public Guid StudentId { get; set; }
        public Guid AdmissionTypeId { get; set; }
        public string RollNo { get; set; }
        public string RefferenceNo { get; set; }

        public string FullName { get; set; }
        public string FatherName { get; set; }
        public string StudentCNIC { get; set; }
        public Guid SectionId { get; set; }
        public string Range { get; set; }
        public bool IsSelected { get; set; }
        public bool IsDisabled { get; set; }
        public Guid GenderId { get; set; }
        public int StatusId { get; set; }
        public string SectionName { get; set; }
        public Guid SessionId { get; set; }
        public Guid CampusId { get; set; }
        public Guid ProgramDetailId { get; set; }

        public Guid SectionCourseLinkId { get; set; }

        [Key]
        public Guid EnrollmentId { get; set; }
        public Guid LoggerId { get; set; }
        public Guid ProgramId { get; set; }
        public string ClassName { get; set; }


    }
    [Table("VWToEnrollWithoutPaid", Schema = "Registration")]
    public partial class ToEnrollWithoutPaidVM
    {
        [Key]
        public Guid AdmissionFormId { get; set; }
        public Guid CampusProgramId { get; set; }
        public Guid ClassId { get; set; }

        public Guid StudentId { get; set; }
        public Guid AdmissionTypeId { get; set; }
        public string RollNo { get; set; }
        public string RefferenceNo { get; set; }

        public string FullName { get; set; }
        public string FatherName { get; set; }
        public string StudentCNIC { get; set; }
        public Guid SectionId { get; set; }
        public string Range { get; set; }
        public bool IsSelected { get; set; }
        public bool IsDisabled { get; set; }
        public Guid GenderId { get; set; }
        public int StatusId { get; set; }
    }


    public class StudentPromotionList
    {
        [Key]
        public Guid AdmissionFormId { get; set; }

        public string RefferenceNo { get; set; }

        public string RollNo { get; set; }


        public string FullName { get; set; }

        public Guid SectionCourseLinkId { get; set; }

        public bool IsSelected { get; set; }


    }

    public class StudentPromotionPreList
    {
        [Key]
        public Guid AdmissionFormId { get; set; }

        public string RefferenceNo { get; set; }

        public string RollNo { get; set; }


        public string FullName { get; set; }

        public Guid SectionCourseLinkId { get; set; }

        public bool IsSelected { get; set; }
        public string Range { get; set; }
        public string NewRollNo { get; set; }





    }
}

File: src/ui/app/root/application/models/Registration/Enrollments.ts
export interface IRegistrationEnrollments {
    enrollmentId: string;
    admissionFormId: string;
    sectionCourseLinkId: string[];
    statusId: number;
    loggerId: string;

}

export interface IStudentToEnrollVM {
    admissionFormId: string;
    campusProgramId: string;
    classId: string;
    studentId: string;
    admissionTypeId: string;
    rollNo: string;
    refferenceNo: string;
    fullName: string;
    fatherName: string;
    studentCNIC: string;
    sectionId: string;
    range: string;
    isSelected: boolean;
    isDisabled: boolean;
    genderId: string;
    statusId: number;
    obtainedMarks:string;

}
export interface IStudentsToEnrolledPercentageVM {
    admissionFormId: string;
    campusProgramId: string;
    classId: string;
    studentId: string;
    admissionTypeId: string;
    rollNo: string;
    refferenceNo: string;
    fullName: string;
    fatherName: string;
    studentCNIC: string;
    sectionId: string;
    range: string;
    isSelected: boolean;
    isDisabled: boolean;
    genderId: string;
    statusId: number;

}

export interface IStudentEnrolledVM {
    admissionFormId: string;
    campusProgramId: string;
    classId: string;
    studentId: string;
    admissionTypeId: string;
    rollNo: string;
    refferenceNo: string;
    fullName: string;
    fatherName: string;
    studentCNIC: string;
    sectionId: string;
    range: string;
    isSelected: boolean;
    isDisabled: boolean;
    genderId: string;
    statusId: number;
    sectionName: string;
    sessionId: string;
    campusId: string;
    programDetailId: string;
    sectionCourseLinkId: string;
    enrollmentId: string
    loggerId: string;
    programId:string;
    className:string;

}

export interface IToEnrollWithoutPaidVM {
    admissionFormId: string;
    campusProgramId: string;
    classId: string;
    studentId: string;
    admissionTypeId: string;
    rollNo: string;
    refferenceNo: string;
    fullName: string;
    fatherName: string;
    studentCNIC: string;
    sectionId: string;
    range: string;
    isSelected: boolean;
    isDisabled: boolean;
    genderId: string;
    statusId: number;
}

export interface HadafStudent {
    StudentId: string;
    FullName: string;
    FatherName: string;
    StudentCNIC: string;
    ParentCNIC: string;
    StudnetContactNo: string;
    ParentContactNo: string;
    GenderId?: string;
    DateOfBirth?: string;
    Address: string;
    BloodGroupId?: string;
    ReligionId?: string;
    StatusId: number;
    LoggerId?: string;
    AcademicInfo: string;
    Images: string;
    AdmissionFormId?: string;
    CampusProgramId?: string;
    AdmissionTypeId?: string;
    GroupID?: string;
    RollNo: string;
    ReferenceNo: string;
    AdmissionDate?: string;
    isSelected: boolean;
}

export interface CmsModel {
    City: string;
    Campus: string;
    Combination: string;
    AdmissionYear: number;
    CurrentCampusID: number;
    Program: string;
    Shift_ID: number;
    Medium_ID: number;
    Admission_Form_ID: number;
    Program_Detail_ID: number;
    Rollno: string;
    ReferenceNo: string;
    AdmissionType: string;
    Admission_Date_Save: string;
    Admission_Date: string;
    Status_ID: number;
    StudentName: string;
    FatherName: string;
    StudentCNIC: string;
    Father_CNIC: string;
    StudentContactNo: string;
    Father_Phone_No: string;
    Guardians: string;
    Date_of_Birth: string;
    Gender_ID: number;
    Blood_Group_ID: string;
    Religion_ID: string;
    Image: string;
    Address: string;
    Class: string;
    Matric_Registration_Number: string;
    Board_Registration_Number: string;
    Academicinfo: string;
    IsSelected: boolean;
}


export interface StudentPromotionList {
    admissionFormId: string;
    refferenceNo: string;
    rollNo: string;
    fullName: string;
    sectionCourseLinkId:string;
    isSelected:boolean;
}
export interface StudentPromotionListEx {
    admissionFormId: string;
    refferenceNo: string;
    rollNo: string;
    fullName: string;
    sectionCourseLinkId:string;
    campusProgramId:string;
    isSelected:boolean;
}

export interface StudentPromotionPreList {
    admissionFormId: string;
    refferenceNo: string;
    rollNo: string;
    fullName: string;
    sectionCourseLinkId:string;
    isSelected:boolean;
    range: string;
    newRollNo :string;
}

